Change developer tier to serverless tier (#10824)
pingcap/docs: tidb-cloud/use-htap-cluster.md

> **Note:**
>

> > Serverless Tier clusters have TiFlash support and can disable TiFlash.

Reviewer: ```suggestion
> Serverless Tier clusters support TiFlash. You can enable or disable TiFlash based on your needs.
```

Author: Oops, it was my bad. TiFlash is always enabled In Serverless Tier, users can not disable it.

Author: Can you take another look at it? I tried to fix it but not sure how to make it more concise and clear.

> > Serverless Tier clusters support TiFlash. You can not disable TiFlash.

Reviewer: ```suggestion
> TiFlash is always enabled for Serverless Tier clusters. You cannot disable it.
```
@sunxiaoguang How about this?

Author: A lot better now. Thanks!
